### PR TITLE
Added hover effect in shop and product page

### LINKED ISSUE #69: Feat: Implement Animations and Enhance User Experience.
Best Effects on Buttons ,
Transition Effects while scrolling,
hover and animations ,
Please assign me this issuse @pooranjoyb 
i'll be happy to resolve it asap.

### PR DESCRIPTION
**Pull request by Mujtabaa07**
Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #69 

## Type of change

Please give a X on it which is applicable

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactor Code
- [ ] A documentation update
- [ ] Others(mentioned in the issue number)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**_Test A Describe here_**

**_Test B Describe here (if Requred)_**

# Screenshorts and Vedios:

Uploading Recording 2024-05-26 160308 zoooommm.mp4…

give screenshorts and vedio of the changes you made

# Checklist:
give a X on it which is applicable

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by pooranjoyb — COMMENTED** [review]
Great job, @Mujtabaa07! 🎉 Thank you for submitting your pull request. Your contribution is valuable and we appreciate your efforts to improve our project.<p>We will promptly review your changes and offer feedback. Keep up the excellent work! Kindly remember to check our [contributing guidelines](https://github.com/pooranjoyb/popShop/blob/dev/CONTRIBUTING.md)</p>

**Comment by pooranjoyb** [comment]
Hey @Mahitej28 could you please review this?

**Comment by Mahitej28** [comment]
> Hey @Mahitej28 could you please review this?

Sure

**Comment by Mahitej28** [comment]
@Mujtabaa07 there are some changes made in the main branch I suggest you take a pull from `dev` so that I can preview it.

**Comment by Mujtabaa07** [comment]
Sure

File: src/pages/Home/ProductStyles.css [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/* ProductStyles.css */
+img {
```
Comment (Mahitej28) [inline src/pages/Home/ProductStyles.css:2]: @Mujtabaa07 why have you created a separate CSS file for the animation, add your elements in `index.css` with proper comments.

**Review by Mahitej28 — CHANGES_REQUESTED** [review]
Do the requested changes

File: src/pages/Home/ProductStyles.css (hunk 1 above)
Comment (Mujtabaa07) [reply src/pages/Home/ProductStyles.css:2]: Sure I'll make changes now

Comment (Mujtabaa07) [reply src/pages/Home/ProductStyles.css]: @Mahitej28 , I have made the requested changes. Please review.

**Review by Mahitej28 — APPROVED** [review]

**Comment by Mahitej28** [comment]
Looks good to me will merge by end of the day!

**Pull request merged**